vql/darwin/xattr_test.go

```diff
@@ -80,23 +57,35 @@ func (self *XattrTestSuite) TestXAttr() {
 		},
 		{
 			name: filepath,
-			attr: []string{"invalud.test"},
+			attr: []string{"invalid.test"},
 			pass: false,
 		},
 	}
 
+	xattr.Set(filepath, attr1, []byte("test-value"))
+	xattr.Set(filepath, attr2, []byte("test-value"))
+
+	self.Ctx = context.Background()
+	sub_ctx, cancel := context.WithCancel(self.Ctx)
+	defer cancel()
+
+	builder := services.ScopeBuilder{
+		Config:     self.ConfigObj,
+		ACLManager: acl_managers.NullACLManager{},
+		Logger:     logging.NewPlainLogger(self.ConfigObj, &logging.FrontendComponent),
+		Env:        ordereddict.NewDict(),
+	}
+
+	manager, err := services.GetRepositoryManager(self.ConfigObj)
+	assert.NoError(self.T(), err)
+
+	scope := manager.BuildScope(builder)
+	defer scope.Close()
+
 	for i, test := range testcases {
 		defer scope.Close()
 		ret := XAttrFunction{}.Call(sub_ctx, scope, ordereddict.NewDict().Set("filename", test.name).Set("attribute", test.attr))
-		if ret == nil {
-			if test.pass {
-				self.T().Errorf("xattr: test %d: Got %t, espected %t", i, false, test.pass)
-			}
-		} else {
-			if !test.pass {
-				self.T().Errorf("xattr: test %d: Got %t, espected %t", i, true, test.pass)
-			}
-		}
+		assert.Equal(self.T(), test.pass, (ret != vfilter.Null{}), "These two values should be the same. Test %d", i)
```

Comment:
maybe instead of comparing the test to null (as a bool) you can serialize the output and compare it to what is expected?